src/integ_test/java/games/strategy/engine/lobby/server/userDB/DbUserControllerIntegrationTest.java

```diff
@@ -15,90 +16,115 @@
 
 public class DbUserControllerIntegrationTest {
 
+
   @Test
   public void testCreate() throws Exception {
+    DbUser user = createUser();
+    int startCount = getUserCount(() -> Database.getDerbyConnection());
+
+    final DbUserController controller = new DbUserController();
+    controller.createUser(user, new HashedPassword(MD5Crypt.crypt(user.name)));
+
+    int endCount = getUserCount(() -> Database.getDerbyConnection());
+    assertEquals(endCount, startCount + 1);
+    assertTrue(controller.doesUserExist(user.name));
+  }
+
+  private static DbUser createUser() {
     final String name = Util.createUniqueTimeStamp();
     final String email = name + "@none.none";
-    final String password = MD5Crypt.crypt(name);
-    final DbUserController controller = new DbUserController();
-    controller.createUser(name, email, password, false);
-    assertTrue(controller.doesUserExist(name));
+    return new DbUser(
+        new DbUser.UserName(name),
+        new DbUser.UserEmail(email));
+  }
 
-    try (final Connection con = Database.getConnection()) {
-      final String sql = " select * from TA_USERS where userName = '" + name + "'";
-      final ResultSet rs = con.createStatement().executeQuery(sql);
-      assertTrue(rs.next());
-      assertEquals(email, rs.getString("email"));
-      assertEquals(password, rs.getString("password"));
-    }
+  private static int getUserCount(final Supplier<Connection> connection) throws Exception {
+    final Connection dbConnection = connection.get();
+    final String sql = "select count(*) from TA_USERS";
+    final ResultSet rs = connection.get().createStatement().executeQuery(sql);
```

Comment:
Should `connection.get()` be `dbConnection`?  Otherwise, it's being orphaned.  Also, might want to consider using `dbConnection` in a try-with-resources block now that you are passing a `Supplier` to this method to avoid the explicit call to `close()` below.